bravado/mapping/http_client.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+#
+# Copyright (c) 2013, Digium, Inc.
+# Copyright (c) 2014, Yelp, Inc.
+#
+
+import urlparse
+"""HTTP client abstractions.
+"""
+
+APP_FORM = 'application/x-www-form-urlencoded'
+APP_JSON = 'application/json'
+MULT_FORM = 'multipart/form-data'
+
+
+class HttpClient(object):
+    """Interface for a minimal HTTP client.
+    """
+
+    def set_basic_auth(self, host, username, password):
+        """Configures client to use HTTP Basic authentication.
+
+        :param host: Hostname to limit authentication to.
+        :param username: Username
+        :param password: Password
+        """
+        raise NotImplementedError(
+            u"%s: Method not implemented", self.__class__.__name__)
+
+    def set_api_key(self, host, api_key, param_name=u'api_key'):
+        """Configures client to use api_key authentication.
+
+        The api_key is added to every query parameter sent.
+
+        :param host: Hostname to limit authentication to.
+        :param api_key: Value for api_key.
+        :param param_name: Parameter name to use in query string.
+        """
+        raise NotImplementedError(
+            u"%s: Method not implemented", self.__class__.__name__)
```

Comment:
The above two methods (`set_basic_auth` and `set_api_key`) are not actually part of the interface. They are never implemented by the fido client (and I don't think they should be).

Can you remove them from this interface class?